index.html

```diff
@@ -654,5 +654,400 @@ <h2>Exposing change of MediaStreamTrack configuration</h2>
     </p>
     </div>
   </section>
+  <section>
+    <h2>Human faces</h2>
+    <p>Human face metadata describes the human faces in video frames. It can
+    be set by web applications using the standard means when creating
+    {{VideoFrameMetadata}} for {{VideoFrame}}s or it can be set by an user agent
+    when the media track constraints, defined below, are used to
+    enable face detection for the {{MediaStreamTrack}} which provides the
+    {{VideoFrame}}s.</p>
+    <p>The facial metadata may be used by video encoders to enhance the quality
+    of the faces in encoded video streams or for other suitable purposes.</p>
+    <section>
+      <h3>{{VideoFrameMetadata}}</h3>
+      <pre class="idl">
+partial dictionary VideoFrameMetadata {
+  sequence&lt;HumanFace&gt; humanFaces;
+};</pre>
+      <section class="notoc">
+        <h4>Members</h4>
+        <dl class="dictionary-members" data-link-for="VideoFrameMetadata" data-dfn-for="VideoFrameMetadata">
+          <dt><dfn><code>humanFaces</code></dfn> of type <span class="idlMemberType"><code>sequence&lt;{{HumanFace}}&gt;</code></span></dt>
+          <dd>
+            <p>The set of known human faces in this video frame.</p>
+          </dd>
+        </dl>
+      </section>
+    </section>
+    <section>
+      <h3>{{HumanFace}}</h3>
+      <pre class="idl">
+dictionary HumanFace {
+  long              id;
+  float             probability;
+  DOMRectReadOnly   boundingBox;
+  HumanFaceLandmark leftEye;
+  HumanFaceLandmark rightEye;
+  HumanFaceLandmark mouth;
+};</pre>
+      <section class="notoc">
+        <h4>Dictionary {{HumanFace}} Members</h4>
+        <dl class="dictionary-members" data-dfn-for="HumanFace" data-link-for="HumanFace">
+          <dt><dfn><code>id</code></dfn> of type <span class="idlMemberType">{{long}}</span></dt>
+          <dd>
+            <p>If set, an unique identifer of a face within a sequence. If the same face can be tracked in multiple
+            frames originating from the same {{MediaStreamTrack}} source, {{id}} is set to the same integer value
+            for the face in all frames.</p>
+            <p>User agent MUST NOT select the assigned value of {{id}} in such a way that the detected faces could
+            be correlated to match in any way between different {{MediaStreamTrack}} objects.</p>
+          </dd>
+          <dt><dfn><code>probability</code></dfn> of type <span class="idlMemberType">{{float}}</span></dt>
+          <dd>
+            <p>If set, a probability value in range (0,1].
```

Comment:
I changed it to `required probability` and range between [0,1]. I would suggest 0 to mean that the probability estimate is not supported or known (more of this below). 0 is not otherwise a valid probability for a detected face, as you say.

In the practical cases I can think of, if user agent could assign `probability` for a face, it would assign some probability also to all other faces. This might not be true between different types of objects, ie., if faces have (nonzero) probability, landmarks might not have any probability estimate (ie. it would be set to zero).

> If the implementation isn't sharing probability at all, then an app would seem to have no choice but to assume that it is 1

I don't agree with this. I think there's a huge difference whether a probability for an event is known or calculated and whether it is not. A web application might want to run a secondary, more accurate detection algorithm for those face candidates which have low or non-existing (zero) probabilities.